internal/trie/node/encode_decode_test.go

```diff
@@ -57,7 +57,9 @@ func Test_Branch_Encode_Decode(t *testing.T) {
 				Key: []byte{5},
 				Children: [16]Node{
 					&Leaf{
-						HashDigest: []byte{0x41, 0x9, 0x4, 0xa},
+						Key:   []byte{9},
+						Value: []byte{10},
+						Dirty: true,
```

Comment:
Can you instead increase that hash digest to be 32 bytes, and add a test case for the inlined leaf in this test?